examples/go/vm-reconfig-via-tag/handler/handler.go

```diff
@@ -0,0 +1,564 @@
+package function
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+
+	handler "github.com/openfaas/templates-sdk/go-http"
+	"github.com/pelletier/go-toml"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/vapi/rest"
+	"github.com/vmware/govmomi/vapi/tags"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+const cfgPath = "/var/openfaas/secrets/vcconfig"
+
+var (
+	vsClt *vsClient // Persist vSphere connection.
+	once  sync.Once // For handleSignal() to be called once.
+	debug bool      // True for verbose logging.
+)
+
+// vcConfig represents the toml vcconfig file
+type vcConfig struct {
+	VCenter struct {
+		Server, User, Password string
+		Insecure               bool
+	}
+}
+
+// cloudEvent is a subsection of a Cloud Event.
+type cloudEvent struct {
+	Data types.Event `json:"data,omitempty"`
+}
+
+type vsClient struct {
+	govmomi *govmomi.Client
+	rest    *rest.Client
+	tagMgr  *tags.Manager
+}
+
+type vmConfig struct {
+	name    string
+	valBool bool
+	valInt  int
+}
+
+// Handle a function invocation
+func Handle(req handler.Request) (handler.Response, error) {
+	ctx := context.Background()
+	debug = setVerbosity()
+
+	// Load config every time, to ensure the most updated version is used.
+	cfg, err := loadConfig(cfgPath)
+	if err != nil {
+		return handlerResponseWithError("load vcconfig: %w",
+			http.StatusInternalServerError,
+			err,
+		)
+	}
+
+	// The Mananged Object Reference for the VM that powered on.
+	vmMOR, err := eventVmMOR(req.Body)
```

Comment:
I think (convention?) would be to call the fn `getEventVmMoref` (feel free to keep though).

Reply:
I believe you are referring to two points: (1) using `get` in a getter function name, and (2) using MoRef abbreviation.

For the first, I try to avoid using `get` in a getter function name because of what was written in Effective Go. Here an excerpt:
>  it's neither idiomatic nor necessary to put Get into the getter's name. If you have a field called owner (lower case, unexported), the getter method should be called Owner (upper case, exported), not GetOwner. The use of upper-case names for export provides the hook to discriminate the field from the method. 

Source: https://golang.org/doc/effective_go.html#Getters

But, I'm not sure if `eventVmMOR` is a getter function. It's not exported. Does that mean maybe I should use `get`? I do feel it's pretty clear that I'm getting the VM MoRef with out using `get`. 

As for the second, using `MOR` vs. `MoRef`, I think it's common to say `MoRef` or write `moref` in conversation, but in the Govmomi code base, I don't really see that used outside of error messages and comments.  In the Govmomi code base, VM morefs have names of `Self`, `This`, `VM` or just `Reference`. 

I think it's pretty clear and clean to use vmMOR, especially because its type is clearly defined as`types.ManagedObjectReference`.

Reply:
Your rational makes sense! Thx for clarification